src/Log/LogService.php

```diff
@@ -417,10 +425,21 @@ public function getDefaultFormat()
             $this->defaultFormat = $format;
         }
 
-        $runId = $this->getRunId();
+        $format = $this->addRunIdToFormat($this->defaultFormat);
 
-        $format = rtrim(preg_replace('/%runId%/', $runId, $this->defaultFormat), PHP_EOL) . PHP_EOL;
+        return $format;
+    }
 
+    /**
+     * Add RunId to format string.
+     *
+     * @param $format string
+     * @returns string
+     */
+    public function addRunIdToFormat($format)
```

Comment:
A more generic name would be better. We might want to support more of our own placeholders later.

Reply:
I am thinking of `replacePlaceholderWithRunId()`. What do you think? Do you have any names in mind? We can also change the arguments to accept a string (format), a placeholder string, and a replacing string. Then it will have support for all kinds of placeholders. However, there will be only one line of code left inside then.

Reply:
The non-generic part I was thinking of is **RunId**. I could create a subclass, replacing that function and process more placeholders than `%runId%`. A straight forward extension of the functionality, except the function would still be called something with **runId**. Hope this helps.

Reply:
About the second part of adding additional arguments. That might be a nice idea in a different situation, but here we want a function that takes care of replacing placeholders in our format string. Our placeholders, the ones, that are not used by Zend. Adding those arguments move the decision of what needs to be replaced outside of this function into the function calling it, which then has to do at least two jobs. Not the right move here.